test/L2/L2Reward.t.sol

```diff
@@ -506,384 +553,196 @@ contract L2RewardTest is Test {
+        address[] memory stakers = given_anArrayOfStakersOfLength(5);
+        uint256[] memory lockIDs = new uint256[](5);
+        uint256[] memory balances = new uint256[](5);
+        uint256[] memory oldBalances = new uint256[](5);
+        uint256 funds = LSK(5000);
+        uint256 balance = LSK(500);
 
-        // stake 13 resumes
-        positionsToBeModified[0] = lockIDs[12];
-        vm.prank(stakers[12]);
-        l2Reward.resumeUnlockingCountdown(positionsToBeModified);
+        // owner gets balance
+        given_accountHasBalance(address(this), funds);
+        // fund the reward contact, 1 LSK per day for 500 days
+        uint256 dailyRewards = given_ownerHasFundedStaking(Funds({ amount: funds, duration: 500, delay: 1 }));
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+        // all the stakers gets balance
+        for (uint8 i = 0; i < stakers.length; i++) {
+            given_accountHasBalance(stakers[i], balance);
+        }
 
-        return (lockIDs, stakers);
+        Scenario memory scenario = Scenario({
+            stakers: stakers,
+            lockIDs: lockIDs,
+            balances: balances,
+            oldBalances: oldBalances,
+            lastClaimedAmount: 0,
+            rewardsSurplus: 0
+        });
+
+        onDay(1);
+        stakerCreatesPosition(0, LSK(150), 150, scenario);
+        stakerCreatesPosition(1, LSK(250), 250, scenario);
+        onDay(10);
+        stakerCreatesPosition(2, LSK(250), 150, scenario);
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(400), 1, 9, scenario);
+        onDay(30);
+        stakerCreatesPosition(3, LSK(50), 100, scenario);
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(650), 10, 29, scenario);
+        onDay(35);
+        stakerCreatesPosition(4, LSK(10), 120, scenario);
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(700), 30, 34, scenario);
+        onDay(40);
+        allStakersClaim(scenario);
+        checkRewardsContractBalance(funds);
+        onDay(50);
+        allStakersClaim(scenario);
+        checkRewardsContractBalance(funds);
+        onDay(90);
+        allStakersClaim(scenario);
+        checkRewardsContractBalance(funds);
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(710), 35, 89, scenario);
+        onDay(130);
+        allStakersClaim(scenario);
+        checkRewardsContractBalance(funds);
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(710), 90, 129, scenario);
+        onDay(150);
+        allStakersClaim(scenario);
+        checkRewardsContractBalance(funds);
+        // stake 3's expDate is on 130th day so total amount locked reduces by LSK(50).
+        verifyRewardsSurplusForLockedAmountBetweenDays(dailyRewards, LSK(660), 130, 149, scenario);
+    }
+
+    function checkConsistencyTotalWeight(uint256 expiryDateOfLongestStake) private {
```

Comment:
I see this function is only called only on `test_scenario1`. Should we added also in the other 2 scenarios and some other random places (similar to `checkConsistencyPendingUnlockDailyUnlocked`)?

Reply:
This is modified, all scenarios have calls to checkConsistencyTotalWeight.